samples/ordered_puzzle/ordered_puzzle.html

```diff
@@ -0,0 +1,39 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <script src="https://aframe.io/releases/1.1.0/aframe.min.js"></script>
+    <script src="https://unpkg.com/aframe-drag-controls@1.0.0/index.js"></script>
+    <script src="components/ordered_puzzle.js"></script>
+    <script src="components/target_box.js"></script>
+  </head>
+  <body>
+    <a-scene background="color: #AAD5D7" cursor="rayOrigin: mouse">
+      <a-entity
+        id="primaryCamera"
+        position="0 -30 15"
+        rotation="0 0 0"
+        camera
+        drag-controls="objects: .draggable"
+        wasd-controls
+      ></a-entity>
+
+      <a-entity id="readRoom" position="0 -30 0">
+        <a-entity
+          id="blackboard"
+          visible="true"
+          geometry="primitive: plane; width: 22; height: 15"
+          material="color: #333333; shader: flat; transparent: false"
+        ></a-entity>
+        <a-entity
+          id="ordered-puzzle"
+          ordered-puzzle='jsonData: {"images": [{"imageSrc" : "./images/pic1.png", "width": 3, "height": 3, "xTarget": -7, "yTarget": 0, "x": 0, "y": 4},
+          {"imageSrc" : "./images/pic2.png", "width": 3, "height": 3, "xTarget": -3.5, "yTarget": 0, "x": -2, "y": -4},
```

Comment:
@aaronabraham311 @ahmedhamodi What are your thoughts here? I am wondering whether it's feasible (technically and logically) for admins to define the target coordinates as well, or we should get what the right order should be, and just define our xTarget, yTarget to be something static on frontend side?
Like as an example, the top row would be the randomized order images, and then below that would always be the static row of "Targets", and the number of targets to show would be = num pieces.

Reply:
This is a good point. I'm not sure how to go about getting the target locations dynamically without making it look off center or weird. Should we just support 3-4 different layouts and have those (for lack of better term) hardcoded? Something like 1x5, 3x3, and 2x2 would be the 3 I think we should support. Perhaps even 1x4.

Then I agree that having the images load at the top of the popup, in any format/location, works well and having the target locations fixed based on the layout configuration. Open to thoughts and suggestions here though.

Reply:
There is a potentially simpler fix for this issue. We just don't have targets. 

In the ticket, I mentioned that the clues should reveal themselves. We really don't need to check order:

> - The user will need to order the 5 pictures in a way that gives a clue
>    - For example: the clue is mouse, the password to get out of the room. The images will be the letters of mouse and once arranged properly, will reveal the clue

Maybe another example would help clarify this. Let's say that one of the clues for a puzzle is something to do with the word 'teapot'. We would have all the pictures for each letter jumbled up, but as the user shuffles it around, they realize that it's the word 'teapot'. There is no need to define targets in this case, because the clue reveals itself when you shuffle it around. This makes things simple technically and we can have a variety of use cases for the admin (jigsaw, word plays, visual order like Vivian's example, etc.). As a general product principle: if we want users to be creative, remove as much structure as possible so they have freedom to play around. 

For now, what Vivian created is great and potentially would help with jigsaw puzzles, but I am seriously thinking whether it is possible to remove the order checking and just have the blackboard with all the images. We can use her order checking for a puzzle like jigsaw.

Reply:
Oh my bad, I misinterpreted the ticket and thought that we wanted the correct answer to be unlocked/checked when placed in the correct order, so I added the targets to verify this. If we want to just have a blackboard that allows users to move around images freely and discover the hidden clue themselves, I can add a separate component to do this.

Reply:
Hmmmm, I see. So in that case, in my opinion, let's keep the order checking in here as it is already implemented. Then when we are integrating with the backend code (aka moving this from `samples/*` to `app/public/*`), we will convert this into 2 different types of puzzles:
1. A puzzle with no targets and no checking. In this case, users won't have a target to go towards, but rather arranging it in the right order will reveal a clue that will help with another puzzle.
2. A puzzle with targets and checking for whether it is `solved`. This will need to be scoped out / discussed further, and have only "allowed" layouts, so we can have a pre-selected target box layout.

Does that make sense?

Reply:
We can leave these for other tickets though. What we have here can be merged in, imo

Reply:
Yeah, that’s for a different ticket/PR for sure. I just wanted to make sure we were on the same page for future tickets of this work.